projects/packages/jetpack-mu-wpcom/src/features/newspack-blocks/index.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * The newspack blocks.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+namespace Automattic\Jetpack\Jetpack_Mu_Wpcom\Newspack_Blocks;
+
+use Automattic\Jetpack\Jetpack_Mu_Wpcom;
+
+define( 'MU_WPCOM_NEWSPACK_BLOCKS', true );
+define( 'NEWSPACK_BLOCKS__BLOCKS_DIRECTORY', Jetpack_Mu_Wpcom::BASE_DIR . 'build/' );
+define( 'NEWSPACK_BLOCKS__PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+
+require_once __DIR__ . '/../../utils.php';
+require_once __DIR__ . '/synced-newspack-blocks/class-newspack-blocks.php';
+require_once __DIR__ . '/synced-newspack-blocks/class-newspack-blocks-api.php';
+require_once __DIR__ . '/synced-newspack-blocks/blocks/homepage-articles/class-wp-rest-newspack-articles-controller.php';
+
+/**
+ * Registers Articles block routes.
+ */
+function register_rest_routes() {
+	$articles_controller = new \WP_REST_Newspack_Articles_Controller();
+	$articles_controller->register_routes();
+}
+add_action( 'rest_api_init', __NAMESPACE__ . '\register_rest_routes' );
+
+/**
+ * Filters block name.
+ *
+ * @param string $name Block name.
+ * @return string
+ */
+function blog_posts_block_name( $name ) {
+	if ( 'newspack-blocks/homepage-articles' === $name ) {
+		return 'a8c/blog-posts';
+	}
+
+	if ( 'newspack-blocks/carousel' === $name ) {
+		return 'a8c/posts-carousel';
+	}
+
+	return $name;
+}
+add_filter( 'newspack_blocks_block_name', __NAMESPACE__ . '\blog_posts_block_name' );
+
+/**
+ * Fix the issue that the first post will be excluded from the homepage.
+ *
+ * @param array $args WP_Query arguments as created by build_articles_query().
+ *
+ * @return array $args Returns the constructed WP_Query arguments.
+ */
+function blog_posts_build_articles_query( $args ) {
```

Comment:
What do you think of extracting this part into a separate PR which fixes https://github.com/Automattic/wp-calypso/issues/69306? So this PR only focuses on migrating code. 🤔

Reply:
I'm fine with both. But I guess https://github.com/Automattic/wp-calypso/issues/69306 might take times to resolve 🙈

Reply:
> might take times to resolve

Sorry you mean that fix does not fix the issue completely? 😱

Reply:
I think the current one is just a workaround to remove the current post from `post__not_in` args if it's not in the query loop. But if we want to fix the issue completely, we have to know why the current post is set outside the query loop and whether it makes sense to avoid setting the current post outside the query loop, etc.

Reply:
I see... to my best judgement, let's fix this outside this PR, otherwise people (in the issue) might be confused why it's fixed in some cases but not in other cases. Or it could also be the case that this introduces unexpected behavior in a live site. 🤔

Reply:
Ok